### PR TITLE
Channelz; Expose New Core API

### PR DESCRIPTION
**Pull request by ncteisen**
Exposes the new core APIs for channels, so that C++ layer may start consuming them

~Build on #16020~

Tracking issue: #15340

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.2% +1.35Ki [None]                                                                               +15.0Ki  +0.2%
      +0.2% +1.32Ki [Unmapped]                                                                           +15.0Ki  +0.2%
      +3.2%     +32 [None]                                                                                     0  [ = ]
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry                            0  [ = ]
  +221%    +801 src/core/lib/channel/channelz_registry.cc                                               +801  +221%
      [NEW]    +299 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels                           +299  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry                            +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry                          +235  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                                 +101  [NEW]
      [NEW]     +44 grpc_channelz_get_channel                                                                +44  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                              +41  [NEW]
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                                  +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                                          +34  [NEW]
       +67%     +33 [Unmapped]                                                                               +33   +67%
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                                 +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                                           +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                                            +8  [NEW]
  +1.7%     +96 src/core/lib/surface/channel.cc                                                          +96  +1.7%
      +8.1%     +94 grpc_channel_create_with_builder                                                         +94  +8.1%
      +1.5%      +2 [Unmapped]                                                                                +2  +1.5%
  +0.1%     +16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                           +16  +0.1%
      +1.4%     +16 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked                       +16  +1.4%

 -------------- SHRINKING                                                                            --------------
  -4.6%     -64 src/core/lib/channel/channelz.cc                                                         -64  -4.6%
      [DEL]    -604 grpc_core::channelz::ChannelNode::RenderJSON                                            -604  [DEL]
     -36.5%    -150 grpc_core::channelz::ChannelNode::ChannelNode                                           -150 -36.5%
      [DEL]      -2 grpc_core::channelz::ChannelNode::PopulateConnectivityState(grpc_json*) [clone .loca      -2  [DEL]

  +0.2% +2.18Ki TOTAL                                                                                +15.9Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +72  +0.0%

  [ = ]       0 TOTAL      +72  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.2% +1.35Ki [None]                                                                               +15.0Ki  +0.2%
      +0.2% +1.32Ki [Unmapped]                                                                           +15.0Ki  +0.2%
      +3.2%     +32 [None]                                                                                     0  [ = ]
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry                            0  [ = ]
  +221%    +801 src/core/lib/channel/channelz_registry.cc                                               +801  +221%
      [NEW]    +299 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels                           +299  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry                            +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry                          +235  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                                 +101  [NEW]
      [NEW]     +44 grpc_channelz_get_channel                                                                +44  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                              +41  [NEW]
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                                  +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                                          +34  [NEW]
       +67%     +33 [Unmapped]                                                                               +33   +67%
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                                 +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                                           +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                                            +8  [NEW]
  +1.7%     +96 src/core/lib/surface/channel.cc                                                          +96  +1.7%
      +8.1%     +94 grpc_channel_create_with_builder                                                         +94  +8.1%
      +1.5%      +2 [Unmapped]                                                                                +2  +1.5%
  +0.1%     +16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                           +16  +0.1%
      +1.4%     +16 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked                       +16  +1.4%

 -------------- SHRINKING                                                                            --------------
  -4.6%     -64 src/core/lib/channel/channelz.cc                                                         -64  -4.6%
      [DEL]    -604 grpc_core::channelz::ChannelNode::RenderJSON                                            -604  [DEL]
     -36.5%    -150 grpc_core::channelz::ChannelNode::ChannelNode                                           -150 -36.5%
      [DEL]      -2 grpc_core::channelz::ChannelNode::PopulateConnectivityState(grpc_json*) [clone .loca      -2  [DEL]

  +0.2% +2.18Ki TOTAL                                                                                +15.9Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +72  +0.0%

  [ = ]       0 TOTAL      +72  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                              FILE SIZE
 ++++++++++++++ GROWING                                                                ++++++++++++++
  +0.2% +1.33Ki [None]                                                                 +20.2Ki  +0.2%
      +0.2% +1.30Ki [Unmapped]                                                             +20.2Ki  +0.2%
      +3.2%     +32 [None]                                                                       0  [ = ]
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry              0  [ = ]
  +355% +1.26Ki src/core/lib/channel/channelz_registry.cc                              +1.26Ki  +355%
      [NEW]    +694 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels             +694  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry              +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry            +235  [NEW]
      [NEW]    +132 grpc_channelz_get_channel                                                 +132  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                   +101  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                +41  [NEW]
       +78%     +38 [Unmapped]                                                                 +38   +78%
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                    +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                            +34  [NEW]
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                   +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                             +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                              +8  [NEW]
  +1.7%     +96 src/core/lib/surface/channel.cc                                            +96  +1.7%
      +8.1%     +94 grpc_channel_create_with_builder                                           +94  +8.1%
      +1.5%      +2 [Unmapped]                                                                  +2  +1.5%
  +0.1%     +16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc             +16  +0.1%
      +1.4%     +16 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked         +16  +1.4%

  +0.2% +2.70Ki TOTAL                                                                  +21.6Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +72  +0.0%

  [ = ]       0 TOTAL      +72  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                              FILE SIZE
 ++++++++++++++ GROWING                                                                ++++++++++++++
  +0.2% +1.33Ki [None]                                                                 +20.2Ki  +0.2%
      +0.2% +1.30Ki [Unmapped]                                                             +20.2Ki  +0.2%
      +3.2%     +32 [None]                                                                       0  [ = ]
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry              0  [ = ]
  +355% +1.26Ki src/core/lib/channel/channelz_registry.cc                              +1.26Ki  +355%
      [NEW]    +694 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels             +694  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry              +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry            +235  [NEW]
      [NEW]    +132 grpc_channelz_get_channel                                                 +132  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                   +101  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                +41  [NEW]
       +78%     +38 [Unmapped]                                                                 +38   +78%
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                    +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                            +34  [NEW]
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                   +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                             +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                              +8  [NEW]
  +1.7%     +96 src/core/lib/surface/channel.cc                                            +96  +1.7%
      +8.1%     +94 grpc_channel_create_with_builder                                           +94  +8.1%
      +1.5%      +2 [Unmapped]                                                                  +2  +1.5%
  +0.1%     +16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc             +16  +0.1%
      +1.4%     +16 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked         +16  +1.4%

  +0.2% +2.70Ki TOTAL                                                                  +21.6Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +72  +0.0%

  [ = ]       0 TOTAL      +72  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                              FILE SIZE
 ++++++++++++++ GROWING                                                                ++++++++++++++
  +0.2% +1.33Ki [None]                                                                 +20.2Ki  +0.2%
      +0.2% +1.30Ki [Unmapped]                                                             +20.2Ki  +0.2%
      +3.2%     +32 [None]                                                                       0  [ = ]
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry              0  [ = ]
  +355% +1.26Ki src/core/lib/channel/channelz_registry.cc                              +1.26Ki  +355%
      [NEW]    +694 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels             +694  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry              +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry            +235  [NEW]
      [NEW]    +132 grpc_channelz_get_channel                                                 +132  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                   +101  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                +41  [NEW]
       +78%     +38 [Unmapped]                                                                 +38   +78%
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                    +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                            +34  [NEW]
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                   +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                             +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                              +8  [NEW]
  +1.7%     +96 src/core/lib/surface/channel.cc                                            +96  +1.7%
      +8.1%     +94 grpc_channel_create_with_builder                                           +94  +8.1%
      +1.5%      +2 [Unmapped]                                                                  +2  +1.5%
  +0.1%     +16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc             +16  +0.1%
      +1.4%     +16 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked         +16  +1.4%

  +0.2% +2.70Ki TOTAL                                                                  +21.6Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +72  +0.0%

  [ = ]       0 TOTAL      +72  +0.0%



```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/lib/channel/channelz_registry.cc [hunk 1]
```diff
@@ -125,3 +125,27 @@ char* ChannelzRegistry::InternalGetTopChannels(intptr_t start_channel_id) {
...
+  grpc_core::channelz::ChannelNode* channel_node =
+      grpc_core::channelz::ChannelzRegistry::GetChannelNode(channel_id);
+  if (channel_node == nullptr) {
+    return nullptr;
+  }
+  grpc_json* top_level_json = grpc_json_create(GRPC_JSON_OBJECT);
+  grpc_json* json = top_level_json;
+  grpc_json* channel_json = channel_node->RenderJson();
+  grpc_json_link_child(json, channel_json, nullptr);
+  channel_json->parent = json;
```
Comment (markdroth) [inline src/core/lib/channel/channelz_registry.cc]: Why are we manually modifying the fields of a json object we just created?

File: include/grpc/grpc.h [hunk 2]
```diff
@@ -162,6 +162,22 @@ GRPCAPI void grpc_completion_queue_thread_local_cache_init(
...
+    you may track the progress by following this github issue:
+    https://github.com/grpc/grpc/issues/15340
+
+    the following APIs return allocated JSON strings that match the response
+    objects from the channelz proto, found here:
+    https://github.com/grpc/grpc/blob/master/src/proto/grpc/channelz/channelz.proto.
+
+    For easy conversion to protobuf, The JSON is formatted according to:
+    https://developers.google.com/protocol-buffers/docs/proto3#json. */
+GRPCAPI char* grpc_channelz_get_top_channels(intptr_t start_channel_id);
```
Comment (markdroth) [inline include/grpc/grpc.h]: Please document the usage and behavior of these methods.

File: src/core/lib/channel/channelz_registry.cc (hunk 1 above)
Comment (ncteisen) [reply src/core/lib/channel/channelz_registry.cc]: Removed the unneeded nulls. Setting `channel_json->key = "channel"` is still needed thought

File: include/grpc/grpc.h (hunk 2 above)
Comment (ncteisen) [reply include/grpc/grpc.h]: Done

File: src/core/lib/channel/channelz_registry.cc (hunk 1 above)
Comment (ncteisen) [reply src/core/lib/channel/channelz_registry.cc]: I have been meaning to make some better convenience functions for our json library now that I have been using it a lot

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                              FILE SIZE
 ++++++++++++++ GROWING                                                                ++++++++++++++
  +0.2% +1.30Ki [None]                                                                 +20.2Ki  +0.2%
      +0.2% +1.27Ki [Unmapped]                                                             +20.2Ki  +0.2%
      +3.2%     +32 [None]                                                                       0  [ = ]
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry              0  [ = ]
  +353% +1.25Ki src/core/lib/channel/channelz_registry.cc                              +1.25Ki  +353%
      [NEW]    +694 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels             +694  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry              +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry            +235  [NEW]
      [NEW]    +124 grpc_channelz_get_channel                                                 +124  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                   +101  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                +41  [NEW]
       +78%     +38 [Unmapped]                                                                 +38   +78%
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                    +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                            +34  [NEW]
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                   +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                             +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                              +8  [NEW]
  +1.7%     +96 src/core/lib/surface/channel.cc                                            +96  +1.7%
      +8.1%     +94 grpc_channel_create_with_builder                                           +94  +8.1%
      +1.5%      +2 [Unmapped]                                                                  +2  +1.5%
  +0.1%     +16 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc             +16  +0.1%
      +1.4%     +16 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked         +16  +1.4%

  +0.2% +2.66Ki TOTAL                                                                  +21.6Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +72  +0.0%

  [ = ]       0 TOTAL      +72  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] Performance differences noted:
Benchmark                                                                                 cpu_time    real_time
----------------------------------------------------------------------------------------  ----------  -----------
BM_PumpStreamClientToServer<InProcess>/32768                                              +9%         +9%
BM_PumpStreamClientToServer<InProcessCHTTP2>/32768                                        +4%         +4%
BM_PumpStreamServerToClient<InProcess>/262144                                             +4%         +4%
BM_StreamingPingPongMsgs<InProcess, NoOpMutator, NoOpMutator>/32768                       +8%         +8%
BM_StreamingPingPongMsgs<MinInProcess, NoOpMutator, NoOpMutator>/32768                    +7%         +7%
BM_StreamingPingPongWithCoalescingApi<InProcess, NoOpMutator, NoOpMutator>/262144/2/1     +9%         +9%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/262144/1/0  +4%         +4%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/262144/2/0  +6%         +6%
BM_StreamingPingPongWithCoalescingApi<MinInProcess, NoOpMutator, NoOpMutator>/262144/2/1  +7%         +7%
BM_UnaryPingPong<InProcess, NoOpMutator, NoOpMutator>/0/262144                            +4%         +4%
BM_UnaryPingPong<InProcess, NoOpMutator, NoOpMutator>/262144/262144                       +5%         +5%
BM_UnaryPingPong<MinInProcess, NoOpMutator, NoOpMutator>/0/2097152                        +5%         +5%
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                                                            FILE SIZE
 ++++++++++++++ GROWING                                                                              ++++++++++++++
  +0.2% +1.27Ki [None]                                                                               +21.2Ki  +0.2%
      +0.2% +1.30Ki [Unmapped]                                                                           +21.2Ki  +0.2%
      [NEW]      +8 grpc_core::channelz::(anonymous namespace)::g_channelz_registry                            0  [ = ]
  +348% +1.24Ki src/core/lib/channel/channelz_registry.cc                                            +1.24Ki  +348%
      [NEW]    +687 grpc_core::channelz::ChannelzRegistry::InternalGetTopChannels                           +687  [NEW]
      [NEW]    +239 grpc_core::channelz::ChannelzRegistry::InternalRegisterEntry                            +239  [NEW]
      [NEW]    +235 grpc_core::channelz::ChannelzRegistry::InternalUnregisterEntry                          +235  [NEW]
      [NEW]    +124 grpc_channelz_get_channel                                                               +124  [NEW]
      [NEW]    +101 grpc_core::channelz::ChannelzRegistry::InternalGetEntry                                 +101  [NEW]
      [NEW]     +41 grpc_core::channelz::ChannelzRegistry::Init                                              +41  [NEW]
      [NEW]     +38 grpc_core::channelz::ChannelzRegistry::ChannelzRegistry                                  +38  [NEW]
      [NEW]     +34 grpc_core::channelz::ChannelzRegistry::Shutdown                                          +34  [NEW]
       +59%     +29 [Unmapped]                                                                               +29   +59%
      [NEW]     +22 grpc_core::channelz::ChannelzRegistry::~ChannelzRegistry                                 +22  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                                                           +21  [NEW]
      [NEW]      +8 grpc_core::channelz::ChannelzRegistry::Default                                            +8  [NEW]
  +1.1%     +64 src/core/lib/surface/channel.cc                                                          +64  +1.1%
      +6.0%     +69 grpc_channel_create_with_builder                                                         +69  +6.0%
  +0.2%     +32 src/core/ext/filters/client_channel/lb_policy/grpclb/grpclb.cc                           +32  +0.2%
      +2.9%     +32 grpc_core::(anonymous namespace)::GrpcLb::ProcessChannelArgsLocked                       +32  +2.9%
  +3.5%     +16 src/core/lib/json/json.cc                                                                +16  +3.5%
       +32%     +12 [Unmapped]                                                                               +12   +32%
      +7.5%      +8 grpc_json_link_child                                                                      +8  +7.5%

 -------------- SHRINKING                                                                            --------------
 -11.7%    -173 src/core/lib/channel/channelz.cc                                                        -173 -11.7%
      [DEL]    -426 grpc_core::channelz::ChannelNode::RenderJSON                                            -426  [DEL]
      [DEL]    -411 grpc_core::channelz::ChannelNode::ChannelNode                                           -411  [DEL]
     -55.5%    -157 grpc_core::channelz::SubchannelNode::SubchannelNode                                     -157 -55.5%
      [DEL]     -69 grpc_core::channelz::ChannelNode::MakeChannelNode                                        -69  [DEL]
      [DEL]      -2 grpc_core::channelz::ChannelNode::PopulateChildRefs(grpc_json*) [clone .localalias.1      -2  [DEL]

  +0.2% +2.45Ki TOTAL                                                                                +22.4Ki  +0.2%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +64  +0.0%

  [ = ]       0 TOTAL      +64  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: src/core/lib/channel/channelz_registry.cc (hunk 1 above)
Comment (ncteisen) [reply src/core/lib/channel/channelz_registry.cc]: I fixed up `grpc_json_link_child` in the descendant PR, and updated this.

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                 FILE SIZE
 ++++++++++++++ GROWING                                   ++++++++++++++
  +0.0%    +331 [None]                                    +1.15Ki  +0.0%
   +10%    +149 src/core/lib/channel/channelz_registry.cc    +149   +10%
      [NEW]    +116 grpc_channelz_get_channel                    +116  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                +21  [NEW]
       +18%     +12 [Unmapped]                                    +12   +18%

  +0.0%    +480 TOTAL                                     +1.30Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +64  +0.0%

  [ = ]       0 TOTAL      +64  +0.0%



```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

File: include/grpc/grpc.h [hunk 3]
```diff
@@ -162,6 +162,29 @@ GRPCAPI void grpc_completion_queue_thread_local_cache_init(
 GRPCAPI int grpc_completion_queue_thread_local_cache_flush(
     grpc_completion_queue* cq, void** tag, int* ok);
 
+/************* CHANNELZ API *************/
```
Comment (markdroth) [inline include/grpc/grpc.h]: Maybe move all of this down to the bottom, so that it's clear that this comment only applies to the two methods.

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE                                                 FILE SIZE
 ++++++++++++++ GROWING                                   ++++++++++++++
  +0.0%    +331 [None]                                    +1.15Ki  +0.0%
   +10%    +149 src/core/lib/channel/channelz_registry.cc    +149   +10%
      [NEW]    +116 grpc_channelz_get_channel                    +116  [NEW]
      [NEW]     +21 grpc_channelz_get_top_channels                +21  [NEW]
       +18%     +12 [Unmapped]                                    +12   +18%

  +0.0%    +480 TOTAL                                     +1.30Ki  +0.0%


****************************************************************

libgrpc++.so

     VM SIZE              FILE SIZE
 ++++++++++++++ GROWIN ++++++++++++++
  [ = ]       0 [None]     +64  +0.0%

  [ = ]       0 TOTAL      +64  +0.0%



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by yashykt** [comment]
Looks like you got the wrong issue :)

**Comment by ncteisen** [comment]
Oops, fixed. thanks!

**Pull request merged**